src/router/routes/exportRoute.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-underscore-dangle */
+import express, { Router } from 'express';
+import { ExportType, InitiateExportRequest, Persistence } from 'fhir-works-on-aws-interface';
+import { BadRequestError } from 'fhir-works-on-aws-interface/lib/errors/BadRequestsError';
+import isString from 'lodash/isString';
+import RouteHelper from './routeHelper';
+import ExportHandler from '../handlers/exportHandler';
+
+export default class ExportRoute {
+    readonly router: Router;
+
+    private exportHandler: any;
+
+    private serverUrl: string;
+
+    constructor(serverUrl: string, persistence: Persistence) {
+        this.router = express.Router();
+        this.serverUrl = serverUrl;
+        // @ts-ignore
+        this.exportHandler = new ExportHandler(persistence);
+        this.init();
+    }
+
+    async initiateExportRequests(req: express.Request, res: express.Response, requestGranularity: ExportType) {
+        const requestQueryParams: any = {};
+        // eslint-disable-next-line no-unused-expressions
+        isString(req.query._outputFormat) ? (requestQueryParams._outputFormat = req.query._outputFormat) : '';
+        // eslint-disable-next-line no-unused-expressions
+        Number(req.query._since) ? (requestQueryParams._since = Number(req.query._since)) : '';
+        // eslint-disable-next-line no-unused-expressions
+        isString(req.query._type) ? (requestQueryParams._type = req.query._type) : '';
+
+        if (requestQueryParams._outputFormat && requestQueryParams._outputFormat !== 'ndjson') {
+            throw new BadRequestError('We only support exporting resources into ndjson formatted file');
```

Comment:
Use `http-errors` instead
```js
import createError from 'http-errors';
throw new createError.BadRequest(message);
```

The routing layer can throw http errors. There's no need to create a custom error like `BadRequestError`

Reply:
Good idea. fixed